### PR TITLE
Fix/login event

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé

Create a LoginEvent when a user is forced to update their password, or requests a password reset, as this process automatically logs the user in. The user will see this even on the Sign-in History page when they log in.

# Test instructions | Instructions pour tester la modification

See cds-snc/notification-planning#615 for more details including test plan.

### CODE REVIEW

**Review by jzbahrai — COMMENTED** [review]
looks good to me but will also want a second review.

**Review by jzbahrai — APPROVED** [review]

File: app/main/views/authenticator.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def __enter__(self) -> User:
 
         # Check if coming from new password page
         if "password" in session.get("user_details", {}):
-            self.user.update_password(session["user_details"]["password"])
+            self.user.update_password(session["user_details"]["password"], session["user_details"]["loginData"])
```
Comment (sastels) [inline app/main/views/authenticator.py:18]: Should also check if `loginData` is a key in `session.get("user_details", {})`

**Review by sastels — CHANGES_REQUESTED** [review]
Code looks good with one minor request. Haven't tested it yet, will do that ASAP.

File: app/main/views/authenticator.py (hunk 1 above)
Comment (andrewleith) [reply app/main/views/authenticator.py:18]: Done.

**Review by sastels — APPROVED** [review]
Tested locally, verified that login_events table is getting new rows for forced password reset. Good job!

**Review by jimleroyer — COMMENTED** [review]
Great job @andrewleith ! 👏

**Pull request merged**